src/map/lua/lua_baseentity.cpp

```diff
@@ -8625,6 +8625,41 @@ inline int32 CLuaBaseEntity::getAllianceSize(lua_State* L)
     return 1;
 }
 
+/************************************************************************
+*  Function: getAllianceID()
+*  Purpose : Returns the id of the alliance/party or player
+*  Example : local allianceId = player:getAllianceID()
+*  Notes   :
+************************************************************************/
+
+inline int32 CLuaBaseEntity::getAllianceID(lua_State* L)
+{
+    DSP_DEBUG_BREAK_IF(m_PBaseEntity == nullptr);
+
+    uint8 id = 0;
+
+    auto PChar = dynamic_cast<CCharEntity*>(m_PBaseEntity);
+
+    if (PChar != nullptr)
```

Comment:
Inline this

`if (CCharEntity* PChar = dynamic_cast<CCharEntity*>(m_PBaseEntity))`

Reply:
should also be const.

Reply:
Ok changed to:
`if (const CCharEntity* PChar = dynamic_cast<CCharEntity*>(m_PBaseEntity))`